<launch>
	<arg name="message_rate" default="20"/> <!-- in Hz -->
	<arg name="ID" default="0"/>
	<arg name="joystick_type" default="XBOX_XS"/>
		
	<node name="joy_node_$(arg ID)" pkg="joy" type="joy_node" output="screen">
		<remap from="joy" to="joy_$(arg ID)"/>
		<param name="dev" value="/dev/input/js$(arg ID)"/>
		<param name="deadzone" value="0.15"/>
		<param name="default_trig_val" value="true"/>
		<param name="coalesce_interval" value="$(eval 1.0 / arg('message_rate'))"/> <!-- in sec -->
		<param name="autorepeat_rate" value="$(arg message_rate)"/>
	</node>
	<node name="joystick_steering_node_$(arg ID)" pkg="joystick_controller" type="joystick_steering.py" output="screen">
		<param name="ID" value="$(arg ID)"/>
		<param name="joy_message_topic_name" value="joy_$(arg ID)"/>
		<param name="joystick_type" value="$(arg joystick_type)"/>
		<param name="child_mode" value="true"/>
		<param name="child_mode_inertia_i1" value="0.76"/>
		<param name="child_mode_inertia_i2" value="0.08"/>
		<rosparam file="$(find joystick_controller)/config/joystick.yaml" command="load"/>
		<rosparam file="$(find joystick_controller)/config/steering_modes.yaml" command="load"/>
	</node>
</launch>
